### PR TITLE
Addition form config features

### PR DESCRIPTION
**Pull request by ciremusyoka**
Adds the following features to form config UI:
- Time stamp columns on table data
- Upload new file button to drafts component
- Enable passing drill down props

part of issue https://github.com/onaio/reveal-frontend/pull/949 and https://github.com/onaio/reveal-frontend/issues/967

### CODE REVIEW

File: packages/opensrp-form-config/src/helpers/utils.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+type StrNum = string | number;
+
+export const formatDate = (stringDate: string): string => {
```
Comment (moshthepitt) [inline packages/opensrp-form-config/src/helpers/utils.ts:3]: Could you please create an issue to move this to a utils package? (Maybe @onaio/utils).  Also, add docstrings.

File: packages/opensrp-form-config/src/components/Releases/index.tsx [hunk 2]
```diff
@@ -164,15 +183,18 @@ const ManifestReleases = (props: ManifestReleasesProps & DefaultProps) => {
 };
 
 /** declear default props */
```
Comment (moshthepitt) [inline packages/opensrp-form-config/src/components/Releases/index.tsx:185]: What does this docstring mean

File: packages/opensrp-form-config/src/components/UploadFile/index.tsx [hunk 3]
```diff
@@ -220,7 +220,7 @@ UploadConfigFile.defaultProps = defaultProps;
  * @param {Partial<Store>} -  the  redux store
  */
 // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unused-vars
-const mapStateToProps = (state: Partial<Store>, ownProps: any): DefaultProps => {
+const mapStateToProps = (state: Partial<Store>, ownProps: any): UploadDefaultProps => {
```
Comment (p-netm) [inline packages/opensrp-form-config/src/components/UploadFile/index.tsx:223]: lets please remove any mention here, 
and also was disabling the `@typescript-eslint/no-unused-vars` rule on the line above necessary

File: packages/opensrp-form-config/src/components/DraftFiles/index.tsx [hunk 4]
```diff
@@ -197,14 +219,15 @@ const ManifestDraftFiles = (props: ManifestDraftFilesProps) => {
...
+            maxWidth: 80,
         },
     ];
 
     const DrillDownTableProps = {
         columns,
         data: stateData,
-        paginate: false,
         useDrillDown: false,
+        ...drillDownProps,
```
Comment (p-netm) [inline packages/opensrp-form-config/src/components/DraftFiles/index.tsx:230]: @ciremusyoka I believe the intention here is that new  or rather propped in `drillDownProps` will override any of the ones already locally defined, yes? including `data`

Comment (ciremusyoka) [reply packages/opensrp-form-config/src/components/DraftFiles/index.tsx:230]: @p-netm Have picked several `drillDownProps` but `data` and `columns` aren't among the picked i.e you can't override data and columns.

File: packages/opensrp-form-config/src/components/Releases/index.tsx (hunk 2 above)
Comment (ciremusyoka) [reply packages/opensrp-form-config/src/components/Releases/index.tsx]: Have made it more clear.

File: packages/opensrp-form-config/src/helpers/utils.ts (hunk 1 above)
Comment (ciremusyoka) [reply packages/opensrp-form-config/src/helpers/utils.ts:8]: issue #263

**Pull request merged**